spec/controllers/pieces_controller_spec.rb

```diff
@@ -76,5 +77,56 @@
       end
     end
 
+    context "check?" do
+      it "does not return an error message if a white piece puts the black king in check" do
```

Comment:
Hi! Yeah, so I wasn't sure how to link the rspec tests to the java alert messages... I will look and see if this is possible. However, if an error message is present, then the piece should not have been able to move. That is why it should be sufficient to test the piece's location after the update....and yes I agree that we should show that the game is in check...I will update that